src/lib/core/mp_datetime.c

```diff
@@ -69,7 +70,7 @@ datetime_unpack(const char **data, uint32_t len, struct datetime *date)
 	if (len == 0)
 		goto return_validated;
 
-	if (len != SZ_TAIL) {
+	if (len != SZ_TAIL || *data + len != end) {
```

Comment:
`*data + len` always equals `end` AFAICS. Why do you need this check?

Reply:
Indeed, this is redundant, looks like I forgot to drop this change — sorry for the inconvenience. Fixed.